farm/train.py

```diff
@@ -188,14 +187,9 @@ def backward_propagate(self, loss, step):
             loss.backward()
 
         if (step + 1) % self.grad_acc_steps == 0:
```

Comment:
Scheduler step should be done after the optimizer step in PyTorch 1.1.0 and later. See the note [in the documentation](https://pytorch.org/docs/stable/optim.html#how-to-adjust-learning-rate).